src/init.cpp

```diff
@@ -1225,6 +1225,19 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
                     LogPrintf("Prune: pruned datadir may not have more than %d blocks; -checkblocks=%d may fail\n",
                         MIN_BLOCKS_TO_KEEP, GetArg("-checkblocks", 288));
                 }
+
+                {
+                    LOCK(cs_main);
+                    CBlockIndex* tip = chainActive.Tip();
+                    CBlock block;
+                    if (tip && ReadBlockFromDisk(block, tip)) {
```

Comment:
Any specific reason to read the block from disk here? Reminder: `tip` itself, a `CBlockIndex`, also has `nTime`